src/microsim/schema/modality/base_modality.py

```diff
@@ -0,0 +1,24 @@
+from typing import Literal
+
+from microsim.schema._base_model import SimBaseModel
+
+MODALITIES = ["widefield", "confocal", "two-photon", "light-sheet"]
+
+
+class MicroscopeModality(SimBaseModel):
```

Comment:
Make sense. I'll close this PR and put this code in a different branch. Since 3d-SIM, as you say, would not have this render() method, just enforcing a uniform API interface seems a better option as in any case, there is not much code duplication between confocal and widefield.